### ISSUE TITLE
Checking APP_* in $_SERVER var isn't suffisant 

### ISSUE THREAD

**Issue by matthieugauthier** [symfony/recipes#331]
In file recipes/symfony/framework-bundle/3.3/public/index.php ,

checking APP_* in $_SERVER var isn't suffisant, in certain docker context, APP_* vars are set in entrypoint and must be checked by $_ENV var.

It would be usefull to check both vars in index.php .

**Comment by Pierstoval** [comment]
Do you have a reproducible  case where `APP_ENV` is set in `$_ENV` and not in `$_SERVER`?

**Comment by matthieugauthier** [comment]
Here : https://github.com/matthieugauthier/dump-sample-diff-env-server-var .

**Comment by Pierstoval** [comment]
In your case, it's because you override the env var in your `entrypoint.sh`. This is not usual, but this is an edge case that has to be fixed 👍 

I proposed symfony/symfony#25693 to see if we can fix this in the framework instead.

For now, as a fallback you can use a hack like this:

```php
if (!isset($_SERVER['APP_ENV'], $_ENV['APP_ENV']) && false === getenv('APP_ENV')) {
    if (!class_exists(Dotenv::class)) {
        throw new \RuntimeException('APP_ENV environment variable is not defined. You need to define environment variables for configuration or add "symfony/dotenv" as a Composer dependency to load variables from a .env file.');
    }
    (new Dotenv())->load(__DIR__.'/../.env');
}

$env = $_SERVER['APP_ENV'] ?? $_ENV['APP_ENV'] ?? getenv('APP_ENV');
```

**Comment by zim32** [comment]
I can't understand. I have APP_ENV **set** to dev. And my .dot file is not loaded because in this scenario.
Bu it must be loaded in dev env.

**Comment by jrjohnson** [comment]
I believe this is resolved in #464
